src/Microsoft.DotNet.Cli.Utils/CommandResolution/DefaultCommandResolver.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class DefaultCommandResolver : ICommandResolver
+    {
+        private IList<ICommandResolver> _orderedCommandResolvers;
+
+        public DefaultCommandResolver() : this(new PackagedCommandSpecCreator()) { }
+
+        public DefaultCommandResolver(IPackagedCommandSpecCreator packagedCommandSpecCreator)
```

Comment:
My thought was to add a AddResolver method and then have an Extension to DefaultCommandResolver or maybe ICompositeCommandResolver called AddDefaultCommandResolvers that would add all of these. Likewise for ScriptsCommandResolver. If you use the extension, you can get rid of one resolver and just keep the CompositeCommandResolver. You can also unit test the list of resolvers adding by passing a mock of ICompositeCommandResolver and checking what was added to it.